### ISSUE TITLE
Screen sleeps/turns off after inactivity

### ISSUE THREAD

**Issue by jami-shahab** [domferr/tilingshell#85]
**Describe the bug**
Does everything perfectly but if left alone for a while while using tiling shell the screen goes black after some inactivity. It's fine if I quickly move my mouse to wake it up but if I wait too long, once woken up all the windows are messed up and put onto one part of one monitor screen. I am using two monitors for reference and have a different layout running on each.  I am on ubuntu 23.04

**To Reproduce**
1. leave screen alone/remain inactive
2. once screen goes dark wait a while without moving the mouse
3. once screen wakes up again the windows are misarranged

**Information (please complete the following):**
 - Tiling Shell version: [version 99]
 - GNOME version: [44.3].

**Comment by domferr** [comment]
Hey @jami-shahab, unfortunately GNOME forgets about where the windows were placed. I'm experiencing the same and as far as I remember I had this behavior even before using Tiling Shell. So, I'm not sure if it is the default behavior of GNOME.

Do you know anything about that? I'm curious since it is an open question since I started Tiling Shell ahahah

**Comment by thyttan** [comment]
Maybe a solution: https://extensions.gnome.org/extension/4736/smart-auto-move/

I use that to auto-place my startup applications the way I want them.

**Comment by domferr** [comment]
@thyttan that's nice! Thank you

**Comment by domferr** [comment]
Hey guys, I have a quick update for you. I figured this out, and in the next update I should have solved this. If you wanna test this out in advance, here are the files! If you do, please give me some feedback, thanks!

- GNOME shell version 42, 43 or 44: [GNOME.42-44.tilingshell@ferrarodomenico.com.zip](https://github.com/user-attachments/files/16202516/GNOME.42-44.tilingshell%40ferrarodomenico.com.zip)
- GNOME shell version 45 or 46: [tilingshell@ferrarodomenico.com.zip](https://github.com/user-attachments/files/16202518/tilingshell%40ferrarodomenico.com.zip)

___

If you don't know how to install, here it is a little guide for you. You need to perform the following steps: overwrite the files into `.local/share/gnome-shell/extensions/tilingshell@ferrarodomenico.com` with the ones in the archive I shared, then log out and log in (or restart the system).

**Comment by C0Nd3Mnd** [comment]
This issue isn't fixed for me yet as stated in the v12.0 release notes:

> * Fix: Windows no longer lose their position after OS suspension. Closes #85

Most windows move ever-so-slighty from their snapped-to position after suspend, with some moving a lot. I have two displays.

**Comment by domferr** [comment]
Hey @C0Nd3Mnd thank you, it was a typo, I updated the release notes!

**Comment by mathiasbosman** [comment]
Not sure if it's the same issue; (apologies for the bump if not the case)

I'm using dual monitors and when I unlock after one of my monitors went to sleep my windows also shift (and usually in the middle of the monitors to make it more annoying :-D)

But as far as I can find online it probably has something to do with how fast monitors wake up after the OS does.

https://bugs.launchpad.net/ubuntu/+source/mutter/+bug/1927948

**Comment by domferr** [comment]
Yeah @mathiasbosman, thank you for sharing, it seems related! I think we should make some tricks with Tiling Shell after screen wakes up. I'll hope to find some time to implement this!

Tiling Shell attaches to each window info about which tile the window was tiled. My idea is to react on resume from sleep, looping for each non minimized and non fullscreened window and placing it to the correct position (and correct monitor...). It _might_ work 😆

**Comment by icaria36** [comment]
Related: Remember Layout based on Monitor #392

**Comment by Marten-Tech** [comment]
I am also having this issue with with the D2P and Tiling shell combo. It occurs when the system wakes, and places all my windows like the bar was still at the top. even though while during normal operation placement works perfect. I dont know which extension is at fault.
